auth/middleware.go

```diff
@@ -87,37 +95,53 @@ func NewMiddleware(oAuthConfig OAuthConfig, options Options) *Middleware {
 	return m
 }
 
-// Authenticate authenticates a request and returns the Token if validation was successful, otherwise error is returned
-func (m *Middleware) Authenticate(r *http.Request) (Token, error) {
+// authenticate authenticates a request and returns the Token and the client certificate if validation was successful,
+// otherwise error is returned
+func (m *Middleware) authenticate(r *http.Request) (Token, *x509.Certificate, error) {
 	// get Token from Header
 	rawToken, err := extractRawToken(r)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
 	token, err := m.parseAndValidateJWT(rawToken)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
+	}
+
+	const forwardedClientCertHeader = "x-forwarded-client-cert"
+	var cert *x509.Certificate
+	cert, err = parseCertString(r.Header.Get(forwardedClientCertHeader))
+	if err != nil {
+		return nil, nil, err
+	}
```

Comment:
Is a certificate from now on mandatory for all requests? No more authentication based on only the certificate?

If yes, is there no transition phase where this needs to be made configurable? Not all used services might adapt at the same time.

Reply:
Its only mandatory in regard of crown jewels, but as always there are always exceptions... 
but this change does not activate the certificate check, its only prepared but not yet done. The thumbprint cert check can be used within callback endpoints.